lib/policies/headers/index.js

```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  policy: require('./headers'),
+  schema: {
+    type: 'object',
+    properties: {
+      headersPrefix: {
+        type: 'string'
+      },
+      forwardHeaders: {
```

Comment:
I saw a comment from you where you explained why this can't be mandatory, but I lost it during the rebase. Could you re-explain that?

Reply:
From my understanding 
if the policy will break if not provided - mandatory 
this one will not break.

Reply:
Your supposition is correct but — what's the point of having a forward policy if you do not provide the list of headers you want to forward? According to the code, it would result in a no operation. It's like silently swallowing the error.

On the other hand, fail fast can help people understand that the policy is misconfigured.

For instance, let's say I misconfigure the policy by writing `forwardHeader` (missing `s`). 

With the current code, I would neither notice the error and I might get hard times understanding I wrote the word incorrectly.

With a fail fast approach, I would see `Missing forwardHeaders` parameters, and probably notice that I made a typo.

Does that make sense?

Reply:
how required here will help ? Schemas are just helpers for API

To answer the usecase you described I've added the log waning that there is no `forwardHeaders`

think of another use case 
you have configured headers, but want to disable 
it is convenient to just comment  the properties of object and get {} as parameter. 

And another 
ExpressJS allows you to write empty (req,res,next)=>next()

There is no problem in tolerating empty input

Reply:
> How required here will help ? Schemas are just helpers for API

Hmm, that shouldn't be correct. Schemas are evaluated before the [condition](https://github.com/ExpressGateway/express-gateway/blob/master/lib/conditions/index.js#L11),[policy](https://github.com/ExpressGateway/express-gateway/blob/master/lib/policies/index.js#L31),or plugin gets installed.

It has nothing to do with the Admin API itself. The only thing the Admin API provides is the possibility to query the schema (for the UI, I guess).

Reply:
>Hmm, that shouldn't be correct.
just try it out, you'll be surprised, EG will start as usual  

requests indeed will fail with something like `policyMiddleware is not a function` (very helpful)

But let's imagine things would work as you described
how it will help for this policy? and what you are trying to achieve ? 

I can still write `forwardHeaders: {}` there is no way schema can validate keys. It will still be empty and policy will run in `(r,res,next)=>next()` mode. 

@kevinswiber Please check this out and tell if required should be part of the scheme. I'll add it if you thing it is needed in the morning

P.S. This schema validation thing requires another iteration and good vortex on how it suppose to work. for now the only working part is admin API validation

Reply:
> requests indeed will fail with something like policyMiddleware is not a function (very helpful)

A warning should be issued — and you're absolutely right, it's not failing fast because it's returning `null` instead of throwing an error or doing something else, and the error message is not clear.

> I can still write forwardHeaders: {} there is no way schema can validate keys. 

I think — but I need to verify — there should be the possibility to verify that a json object is not empty — but in this case I do not care, because it has been set empty explicitly from the user, and not considered empty by the gateway itself.

> This schema validation thing requires another iteration and good vortex on how it suppose to work. for now the only working part is admin API validation

Totally true. If **this** is the reason why you want to hold on then I'm fine with reiterating later. I've added a point in the relative issue. Let's merge and 🚀 this.